### DISCUSSION TITLE
[BUG] lockup termination releases immediately

### DISCUSSION THREAD

**Discussion by alexaunter** [galactic3/ft-lockup-contract#11] [Q&A] [answered]
tl;dr
Steps to reproduce:
1. create lockup with vesting schedule partly in the past and lockup schedule in the future
2. terminate immediately
Expected result:
vested amount is available according to lockup schedule in the future
Actual result:
vested amount is available immediately
Root cause:
invalid timestamp set when adjusting lockup schedule: need to add add prev_checkpoint.timestamp
https://github.com/NearDeFi/ft-lockup-contract/blob/1dfe5bdc915ce95799b03bc2c7f6c4d66cb95b71/src/schedule.rs#L124

Demo on testnet:
https://github.com/galactic3/ft-lockup-contract/commit/1d25d5ce05309a3fbb1980581ab1f1ad10abd36f

Proposed fix:
https://github.com/galactic3/ft-lockup-contract/commit/4b8e355623538fcf79f5ce5b413128c88526c4fd

**Comment by frol** [comment, marked as answer]
Good catch! I would immediately create a PR to the repo without even creating a discussion on this fork

**Comment by alexaunter** [comment]
https://github.com/NearDeFi/ft-lockup-contract/pull/2
FYI @frol
